### PR TITLE
Make unnecessary API private

### LINKED ISSUE #315: Make API private
This includes:
- Only having the required variables in `__init__.py` - for example by removing/renaming `version_file`
- Marking files as private

### PR DESCRIPTION
**Pull request by LilianBoulard**
Fixes #315

### CODE REVIEW

**Comment by GaelVaroquaux** [comment]
This is not necessary for release 0.3. We can move it to later.

**Review by jovan-stojanovic — APPROVED** [review]

File: RELEASE_PROCESS.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ here are the main steps and appropriate resources:
 
 1. Update `dirty_cat/CHANGES.rst`. It should be updated at each PR,
    but double-checking before the release is good practice.
-2. Update `dirty_cat/dirty_cat/VERSION.txt` with the new version number
+2. Update `dirty_cat/dirty_cat/_VERSION.txt` with the new version number
```
Comment (GaelVaroquaux) [inline RELEASE_PROCESS.md:20]:
```suggestion
2. Update `dirty_cat/dirty_cat/VERSION.txt` with the new version number
```
I don't think that VERSION.text should be made private.

File: dirty_cat/__init__.py [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
+from ._gap_encoder import GapEncoder
+from ._minhash_encoder import MinHashEncoder
+from ._similarity_encoder import SimilarityEncoder
+from ._super_vectorizer import SuperVectorizer
+from ._target_encoder import TargetEncoder
 
-version_file = os.path.join(os.path.dirname(__file__), "VERSION.txt")
-with open(version_file) as fh:
-    __version__ = fh.read().strip()
+with open(_Path(__file__).parent / "_VERSION.txt") as _fh:
```
Comment (GaelVaroquaux) [inline dirty_cat/__init__.py:13]:
```suggestion
with open(_Path(__file__).parent / "VERSION.txt") as _fh:
```

File: doc/conf.py [hunk 3]
```diff
@@ -69,7 +69,7 @@
 # built documents.
 #
 # The short X.Y version.
-version_file = os.path.join('..', 'dirty_cat', 'VERSION.txt')
+version_file = os.path.join('..', 'dirty_cat', '_VERSION.txt')
```
Comment (GaelVaroquaux) [inline doc/conf.py:72]:
```suggestion
version_file = os.path.join('..', 'dirty_cat', 'VERSION.txt')
```

File: setup.py [hunk 4]
```diff
@@ -4,7 +4,7 @@
 from setuptools import setup, find_packages
 
 version_file = os.path.join(
-    os.path.dirname(__file__), 'dirty_cat', 'VERSION.txt')
+    os.path.dirname(__file__), 'dirty_cat', '_VERSION.txt')
```
Comment (GaelVaroquaux) [inline setup.py:7]:
```suggestion
    os.path.dirname(__file__), 'dirty_cat', 'VERSION.txt')
```

File: setup.py [hunk 5]
```diff
@@ -37,7 +37,7 @@
           ],
           platforms='any',
           packages=find_packages(),
-          package_data={'dirty_cat': ['VERSION.txt']},
+          package_data={'dirty_cat': ['_VERSION.txt']},
```
Comment (GaelVaroquaux) [inline setup.py:40]:
```suggestion
          package_data={'dirty_cat': ['VERSION.txt']},
```

**Review by GaelVaroquaux — CHANGES_REQUESTED** [review]
Only one comment: VERSION.txt should not be made private

**Pull request merged**